src/index.ts

```diff
@@ -20,7 +20,7 @@ try {
         version: packageInfo.version,
     });
 
-    const telemetry = Telemetry.create(session, config);
+    const telemetry = await Telemetry.create(session, config);
```

Comment:
```suggestion
    const telemetry = Telemetry.create(session, config);
```
the creation logic does not be turned async.

I'd say the main change needed is to add `this.commonProperties.is_container_env` intialization into `start` (possibly rename it to `setup` as that naming seems confusing)